internal/web/context.go

```diff
@@ -0,0 +1,22 @@
+package web
+
+import (
+	"github.com/labstack/echo/v4"
+	"go.uber.org/zap"
+)
+
+type AppContext struct {
+	echo.Context
+	AppLogger *zap.Logger
```

Comment:
Instead of embedding echo.Context directly, consider using a pointer to avoid potential issues with method receivers that expect the original context type. This will ensure compatibility when passing around the context.